src/main/java/com/ecore/roles/web/RolesApi.java

```diff
@@ -2,18 +2,23 @@
 
 import com.ecore.roles.web.dto.RoleDto;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.util.UriComponentsBuilder;
 
 import java.util.List;
 import java.util.UUID;
 
 public interface RolesApi {
 
     ResponseEntity<RoleDto> createRole(
-            RoleDto role);
+            RoleDto role,
+            UriComponentsBuilder uriBuilder);
 
     ResponseEntity<List<RoleDto>> getRoles();
 
     ResponseEntity<RoleDto> getRole(
             UUID roleId);
 
+    ResponseEntity<List<RoleDto>> getRolesByUserAndTeam(
```

Comment:
This method has been created to return the list of roles by userId and teamId raised by business rule.